nixos/modules/services/network-filesystems/tahoe/default.nix

```diff
@@ -0,0 +1,229 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+let
+  cfg = config.services.tahoe;
+
+  # The tahoe.cfg is ini-format.
+  settingsFormat = pkgs.formats.ini { };
+
+  # Get some helpers for generating the configuration.
+  utils = import ./utils.nix { inherit config pkgs lib settingsFormat; };
+
+  # These are the node options with special support.  They are a mix of
+  # required options and options that were declared by this module in the past
+  # and are kept for compatibility.
+  nodeOptions = {
+    nickname = mkOption {
+      type = types.str;
+      description = ''
+        The nickname of this Tahoe node.
+      '';
+    };
+    tub.port = mkOption {
+      default = 3457;
+      type = types.int;
+      description = ''
+        The port on which the tub will listen.
+
+        This is the correct setting to tweak if you want Tahoe's storage
+        system to listen on a different port.
+      '';
+    };
+    tub.location = mkOption {
+      default = null;
+      type = types.nullOr types.str;
+      description = ''
+        The external location that the node should listen on.
+
+        This is the setting to tweak if there are multiple interfaces
+        and you want to alter which interface Tahoe is advertising.
+
+        If specified, the port should be included.
+      '';
+      };
+    web.port = mkOption {
+      default = 3456;
+      type = types.int;
+      description = ''
+        The port on which the Web server will listen.
+
+        This is the correct setting to tweak if you want Tahoe's WUI to
+        listen on a different port.
+      '';
+    };
+    client.introducer = mkOption {
+      type = types.nullOr types.str;
+      description = ''
+        The furl for a Tahoe introducer node.
+
+        Like all furls, keep this safe and don't share it.
```

Comment:
is it a problem that this string will end up world readable in the nix store?

Reply:
arguably ...

:cry:

Reply:
Of course, this is a pre-existing defect that I am not making better or worse...

I could do another PR that deprecates these furl options in favor of something better (maybe furlFile options?  but I should think about it some more)

Reply:
Yes this can be handled in a follow-up. I just wanted to make sure you noticed, as you seem to care about tahoe.